DigitalOffice/Services/CheckRightsService/Broker/Consumers/CheckIfUserHasRightConsumer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentValidation;
+using LT.DigitalOffice.Broker.Requests;
+using LT.DigitalOffice.CheckRightsService.Repositories.Interfaces;
+using LT.DigitalOffice.Kernel.Broker;
+using MassTransit;
+
+namespace LT.DigitalOffice.CheckRightsService.Broker.Consumers
+{
+    public class CheckIfUserHasRightConsumer : IConsumer<ICheckIfUserHasRightRequest>
+    {
+        private readonly IValidator<ICheckIfUserHasRightRequest> validator;
+        private readonly ICheckRightsRepository repository;
+
+        public CheckIfUserHasRightConsumer(IValidator<ICheckIfUserHasRightRequest> validator, ICheckRightsRepository repository)
+        {
+            this.validator = validator;
+            this.repository = repository;
+        }
+
+        public async Task Consume(ConsumeContext<ICheckIfUserHasRightRequest> context)
+        {
+            try
+            {
+                validator.ValidateAndThrow(context.Message);
+                await context.RespondAsync<IOperationResult<bool>>(new
```

Comment:
Why validate? This do it on side the sender.

Reply:
The sender validates the response received from the client. I'm validating the request

Reply:
The broker is spinning inside our infrastructure, which means we can trust these requests

Reply:
Yes, I agree with @djino97

Reply:
fixed